src/main.c

```diff
@@ -92,7 +108,7 @@ void timer0_isr() __interrupt 1 __using 1
 {
     // display refresh ISR
     // cycle thru digits one at a time
-    uint8_t digit = displaycounter % 4; 
+    uint8_t digit = displaycounter & 3; 
```

Comment:
Interesting trick - I assume this produces a few less instructions than modulus;)
I assume this should achieve a result similar to mod % x == 0 (i.e., only running once per x counts).